src/gui/src/heatMapRUDY.cpp

```diff
@@ -0,0 +1,125 @@
+//////////////////////////////////////////////////////////////////////////////
+// BSD 3-Clause License
+//
+// Copyright (c) 2023, The Regents of the University of California
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+// * Redistributions of source code must retain the above copyright notice, this
+//   list of conditions and the following disclaimer.
+//
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution.
+//
+// * Neither the name of the copyright holder nor the names of its
+//   contributors may be used to endorse or promote products derived from
+//   this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+#include "heatMapRUDY.h"
+
+#include "odb/db.h"
+
+namespace gui {
+
+RUDYDataSource::RUDYDataSource(utl::Logger* logger, odb::dbDatabase* db)
+    : gui::HeatMapDataSource(logger, "RUDY", "RUDY", "RUDY")
```

Comment:
I am concerned that many users will not know about RUDY or what it stands for.  I think it is fine as an internal name for EDA developers in the code.

We already have 'Routing Congestion' heatmap so it needs to be distinguished from that.  Perhaps 'Estimated Congestion'?

@gadfort @rovinski @oharboe any user input?

Reply:
I think I perefer "Esimated Congestion", though that leaves the question as to why it is estimated and not actual. Also, what kind of congestion? Placement? Routing?

Reply:
Perhaps "Fast routing congestion estimate"?

Reply:
It is estimated from placement without actually running a router.  It can be used earlier in the flow.  My goal is to replace the calls to grt from gpl with it.

I'm afraid long names will get truncated in the GUI as the docking window is usually pretty narrow.

Reply:
Maybe "Estimated Congestion (RUDY)"?
Does the GUI support tooltips? Maybe more info could be added to a tooltip.

Reply:
Why would the user want to use this? Perhaps a name that is relevant to the user?

"Congestion Preview"?

+ manual for details...

Reply:
Here are my suggestions. Go one of two ways

1. Merge RUDY with the existing routing congestion option. Run Rudy first and display the result, and in the background calculate the grt solution then replace the heat map when complete.

2. Merge with existing routing congestion option, and in the double click options panel add a new drop down called algorithm. And include RUDY (fast) and GRT (Accurate). Choose a default, and then let a user choose which one they want.

Reply:
A user might want to use it if running GRT would take a long time. I imagine this would also work before DPL is run.

Reply:
> Here are my suggestions. Go one of two ways
> 
> 1. Merge RUDY with the existing routing congestion option. Run Rudy first and display the result, and in the background calculate the grt solution then replace the heat map when complete.
> 2. Merge with existing routing congestion option, and in the double click options panel add a new drop down called algorithm. And include RUDY (fast) and GRT (Accurate). Choose a default, and then let a user choose which one they want.

I like 1 best. Fast preview is an implementation detail of usability...

Reply:
I think "Estimated congestion" would be a good choice. It is clear to the user what it is and avoids confusion with the routing congestion data.